spec/system/food_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'rails_helper'
+
+RSpec.describe 'Foods', type: :system do
+  describe 'index page' do
+    before(:each) do
+      Food.create(name: 'Mango', measurement_unit: 'pieces', price: 4)
+      Food.create(name: 'Fruit', measurement_unit: 'pieces', price: 8)
+      Food.create(name: 'Apple', measurement_unit: 'pieces', price: 7)
+      user = User.create(email: 'test@example.com', password: 'password')
+      sign_in user
+      visit foods_path
+    end
+
+    let(:foods) { Food.all }
+
+    it 'shows the right name' do
+      foods.each do |food|
+        expect(page).to have_content(food.name)
+      end
+    end
+
+    it 'shows the correct unit' do
+      foods.each do |food|
+        expect(page).to have_content(food.measurement_unit)
+      end
+    end
+
+    it 'shows the correct price' do
+      foods.each do |food|
```

Comment:
- [ ] There are a lot of the integration tests you added that didn't work, Please fix that and make sure that all tests are passing. 

![image](https://github.com/Cosmosha/recipe-app/assets/68796232/cc9cd1bd-821d-403d-9693-d648690581e9)